src/build.sh

```diff
@@ -94,6 +98,16 @@ function kernel_build()
 
   command="make -j$parallel_cores ARCH=$arch $cross_compile"
 
+  if [ -n "$pkg" ]; then
+    revision=$(cat .version) # Has to be gathered before building
+    pkg_rule="$(pkg_get_rule "$pkg")"
+    if [[ "$?" == 95 ]]; then
+      complain "Package type '$pkg' not supported"
+      return 95 # ENOTSUP
+    fi
+    command="$command $pkg_rule"
```

Comment:
I suppose that the pkg magic happens here, right? How about creating a function named `create_kernel_pkg`? When I was reading your code, I felt that this deb package feature was a bit hidden; a function dedicated to that will make the code easier to read.

Reply:
Additionally, you need something to check if we can create the target package in the host system. For example, If I try to use this command in ArchLinux, it will fail due to the lack of a deb package. For now, you can add a simple check or warning that lets the user know that this is only available for Debian-based systems. Finally, if the host is Debian-based, we need to check if the user has the build package.

Reply:
The ideia of using a command like this and not a dedicated function was to use the same code for choosing the number of threads and statistics manager as the main build function. If we use a function for that, there will be code duplication. What do you think?

Reply:
> Additionally, you need something to check if we can create the target package in the host system. For example, If I try to use this command in ArchLinux, it will fail due to the lack of a deb package.

you mean for the lack of the dpkg command, right?

> For now, you can add a simple check or warning that lets the user know that this is only available for Debian-based systems. Finally, if the host is Debian-based, we need to check if the user has the build package.

Hmmm what build package? For the build part, I think it'd suffice to check whether dpkg is a valid command, right? For the deploy part, there are more complications...

Reply:
> The ideia of using a command like this and not a dedicated function was to use the same code for choosing the number of threads and statistics manager as the main build function. If we use a function for that, there will be code duplication. What do you think?

Are you sure that we can use the same command? For example, for generating an RPM package, do you know if we need something else?
In general, I prefer to create a new function to make things easier to read, test, and document.

Reply:
> > Additionally, you need something to check if we can create the target package in the host system. For example, If I try to use this command in ArchLinux, it will fail due to the lack of a deb package.
> 
> you mean for the lack of the dpkg command, right?

yes

> > For now, you can add a simple check or warning that lets the user know that this is only available for Debian-based systems. Finally, if the host is Debian-based, we need to check if the user has the build package.
> 
> Hmmm what build package? For the build part, I think it'd suffice to check whether dpkg is a valid command, right? For the deploy part, there are more complications...

Actually, the deploy part should be easier because we always identify the target distro. If the target is not a Debian-based system, and the user is trying to deploy a deb file, you can block the process.

Reply:
> Are you sure that we can use the same command? For example, for generating an RPM package, do you know if we need something else?

The thing is it's not an actual command, it's an argument to the make command already being built by kw build. This approach allows for the creation of the following packages:

```
  rpm-pkg             - Build both source and binary RPM kernel packages
  binrpm-pkg          - Build only the binary kernel RPM package
  deb-pkg             - Build both source and binary deb kernel packages
  bindeb-pkg          - Build only the binary kernel deb package
  snap-pkg            - Build only the binary kernel snap package
                        (will connect to external hosts)
  dir-pkg             - Build the kernel as a plain directory structure
  tar-pkg             - Build the kernel as an uncompressed tarball
  targz-pkg           - Build the kernel as a gzip compressed tarball
  tarbz2-pkg          - Build the kernel as a bzip2 compressed tarball
  tarxz-pkg           - Build the kernel as a xz compressed tarball
  perf-tar-src-pkg    - Build perf-5.13.0-rc6.tar source tarball
  perf-targz-src-pkg  - Build perf-5.13.0-rc6.tar.gz source tarball
  perf-tarbz2-src-pkg - Build perf-5.13.0-rc6.tar.bz2 source tarball
  perf-tarxz-src-pkg  - Build perf-5.13.0-rc6.tar.xz source tarball
```

(taken from kernel's `make help`)